[DAL] Extend algorithm anatomy (#212)
oneapi-src/oneAPI-spec: source/elements/oneDAL/source/programming_model/end_to_end_example.rst

==================
End-to-end example
==================
Below you can find a typical workflow of using oneDAL algorithm on GPU. The
example is provided for :txtref:`Principal Component Analysis algorithm (PCA)
<alg_pca>`.

The following steps depict how to:

- Read the data from CSV file
- Run the training and inference operations for PCA
- Access intermediate results obtained at the training stage

> #. Include the following header. This make all oneDAL declarations available.

Reviewer: ```suggestion
#. Include the following header that makes all oneDAL declarations available.
```

or

```suggestion
#. Include the following header. This makes all oneDAL declarations available.
```

   .. code-block::

      #include "oneapi/dal.hpp"

      #include <cassert>
      #include <iostream>

#. Create a DPC++ queue with the desired device selector. In this case,
   GPU selector is used:

   .. code-block:: cpp

      const auto queue = sycl::queue{ sycl::gpu_selector{} };

> #. All oneDAL declarations are in the ``oneapi::dal`` namespace,
>    so import all declarations from the ``oneapi`` namespace to use ``dal``
>    instead of ``oneapi::dal`` for brevity:

Reviewer: ```suggestion
#. Since all oneDAL declarations are in the ``oneapi::dal`` namespace,
   import all declarations from the ``oneapi`` namespace to use ``dal``
   instead of ``oneapi::dal`` for brevity:
```

   .. code-block:: cpp

      using namespace oneapi;

> #. Read the data from the CSV file into :txtref:`table` by means of
>    :txtref:`CSV data source <csv-data-source>`:

Reviewer: maybe like this?
```suggestion
#. Use :txtref:`CSV data source <csv-data-source>` to read the data
   from the CSV file into a :txtref:`table`:
```

   .. code-block:: cpp

      const auto data = dal::read<dal::table>(queue, dal::csv::data_source{"data.csv"});

> #. Create an :txtref:`PCA <alg_pca>` descriptor, configure its parameters and
>    run the training on the data loaded from CSV.

Reviewer: ```suggestion
#. Create a :txtref:`PCA <alg_pca>` descriptor, configure its parameters, and
   run the training algorithm on the data loaded from CSV.
```
